cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -326,46 +327,52 @@ public Future<CruiseControlResponse> getUserTaskStatus(String host, int port, St
                                 String userTaskID = response.result().getHeader(USER_TASK_ID_HEADER);
                                 response.result().bodyHandler(buffer -> {
                                     JsonObject json = buffer.toJsonObject();
-                                    JsonObject jsonUserTask = json.getJsonArray("userTasks").getJsonObject(0);
-                                    // This should not be an error with a 200 status but we play it safe
-                                    if (jsonUserTask.containsKey(CC_REST_API_ERROR_KEY)) {
-                                        result.fail(new CruiseControlRestException(
-                                                "Error for request: " + host + ":" + port + path + ". Server returned: " +
-                                                        json.getString(CC_REST_API_ERROR_KEY)));
-                                    }
```

Comment:
What if this is some different error? Do we really want to loop on every single error?